File: step0090/domain-core/src/main/java/de/conciso/starter/PersonenService.java
package de.conciso.starter;

import java.util.Optional;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

// todo: Test zufügen
@Service
public class PersonenService implements Personen {

  private static final Logger log = LoggerFactory.getLogger(PersonenService.class);

  private final PersonDAO personDAO;

  public PersonenService(PersonDAO personDAO) {
    this.personDAO = personDAO;
  }

  @Override
  public Person create(String vorname, String name) {
    log.info("create person: " + vorname + " " + name);
    var person = new Person(vorname, name);
    return personDAO.save(person);
  }

  @Override
  public Optional<Person> findById(int id) {
    log.info("looking for person with id: " + id);
    var found = personDAO.findById(id);
    if (found.isEmpty()) {
      log.warn("no person found with id: " + id);
    }
    return found;
  }
}

File: python0110/personen_service.py
import logging

from models import Person
from person_dao import PersonDAO
from personen import Personen


class PersonenService(Personen):
    _log = logging.getLogger(__qualname__)

    def __init__(self, dao: PersonDAO) -> None:
        self._dao = dao

    def create(self, vorname: str, name: str) -> Person:
        self._log.info("create person: %s %s", vorname, name)
        return self._dao.save(vorname, name)

    def find_by_id(self, id: int) -> Person | None:
        self._log.info("looking for person with id: %d", id)
        found = self._dao.find_by_id(id)
        if found is None:
            self._log.warning("no person found with id: %d", id)
        return found
